R/Connection.R

```diff
@@ -647,16 +717,90 @@ odbcListDrivers <- function(keep = getOption("odbc.drivers_keep"), filter = getO
 
 #' List Available Data Source Names
 #'
-#' List the available data sources on your system. See the [DSN Configuration
-#' files](https://github.com/r-dbi/odbc#dsn-configuration-files) section of the
-#' package README for details on how to install data sources for the most
-#' common databases.
+#' @description
+#'
+#' Collect information about the available data source names (DSNs).
 #'
-#' @return A data frame with two columns.
+#' @return A data frame with two columns:
 #' \describe{
-#'   \item{name}{Name of the data source}
-#'   \item{description}{Data Source description}
+#'   \item{name}{Name of the data source. The entries in this column can be
+#'     passed to the `dsn` argument of [dbConnect()].}
+#'   \item{description}{Data source description.}
 #' }
+#'
+#' @section Configuration:
+#'
+#' This function interfaces with the driver manager to collect information
+#' about the available data source names.
+#'
+#' For **MacOS and Linux**, the odbc package supports the unixODBC driver
+#' manager. unixODBC looks to the `odbc.ini` _configuration file_ for information
+#' on DSNs. Install unixODBC with `brew install unixodbc`, and find the
+#' location(s) of your `odbc.ini` file(s) with `odbcinst -j`.
+#'
+#' In this example `odbc.ini` file:
+#'
+#' ```
+#' [MySQL]
+#' Driver = MySQL Driver
+#' Database = test
+#' Server = 127.0.0.1
+#' User = root
+#' password = root
+#' Port = 3306
+#' ```
+#'
+#' ...the data source name is `MySQL`, which will appear in the `name`
+#' column of this function's output. To pass the DSN as the `dsn` argument to
+#' [dbConnect()], pass it as a string, like `"MySQL"`.
+#' `Driver = MySQL Driver` references the driver `name` in [odbcListDrivers()]
+#' output.
+#'
+#' **Windows** is [bundled](https://learn.microsoft.com/en-us/sql/odbc/admin/odbc-data-source-administrator)
+#' with an ODBC driver manager.
+#'
+#' When a DSN is configured with a driver manager, information on the DSN will
+#' be automatically passed on to [dbConnect()] when its `dsn` argument is set.
+#'
+#' For example, with the `MySQL` data source name configured, and the driver
+#' name `MySQL Driver` appearing in [odbcListDrivers()] output, the code:
+#'
+#' ```
+#' con <-
+#'   dbConnect(
+#'     odbc::odbc(),
+#'     "MySQL",
+#'     Driver = "MySQL Driver",
+#'     Database = "test",
+#'     Server = "127.0.0.1",
+#'     User = "root",
+#'     password = "root",
+#'     Port = 3306
+#'   )
+#' ```
+#'
+#' ...can be written:
+#'
+#' ```
+#' con <- dbConnect(odbc::odbc(), "MySQL")
```

Comment:
```suggestion
#' con <- dbConnect(odbc::odbc(), dsn = "MySQL")
```

I think it's better to spell this out by name

Reply:
[c99a80b](https://github.com/r-dbi/odbc/pull/649/commits/c99a80b2dd361ccecc5bbce11c63be7fc5f3e514).